src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={event => {
+              dispatch(
+                filterSlice.actions.setStatus(event.target.value as Status),
```

Comment:
Ensure that the `Status` type includes 'all', 'active', and 'completed' as possible values. If not, this could lead to type errors.